### PR TITLE
KAFKA-5656: Support bulk attributes request on KafkaMbean where some attributes do not exist

### PR DESCRIPTION
**Pull request by ErikKringen**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6363/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6348/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by ijuma** [comment]
Thanks for the PR. Could you add a test as well, please?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6595/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6580/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by omkreddy** [comment]
LGTM

**Comment by ErikKringen** [comment]
@ijuma Tests added

File: clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java [hunk 1]
```diff
@@ -191,15 +191,15 @@ public Object getAttribute(String name) throws AttributeNotFoundException, MBean
...
+            AttributeList list = new AttributeList();
+            for (String name : names) {
+                try {
                     list.add(new Attribute(name, getAttribute(name)));
-                return list;
-            } catch (Exception e) {
-                log.error("Error getting JMX attribute: ", e);
-                return new AttributeList();
+                } catch (Exception e) {
+                    log.warn("Error getting JMX attribute: ", e);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java:199]: Added the attribute name here before merging.

File: clients/src/test/java/org/apache/kafka/common/metrics/KafkaMbeanTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        assertEquals(2.5, sumAttribute);
+    }
+
+    @Test
+    public void testGetAttributeUnknown() throws Exception {
+        sensor.record(2.5);
+        Object counterAttribute = getAttribute(countMetricName);
+        assertEquals(1.0, counterAttribute);
+        Object sumAttribute = getAttribute(sumMetricName);
+        assertEquals(2.5, sumAttribute);
```
Comment (ijuma) [inline clients/src/test/java/org/apache/kafka/common/metrics/KafkaMbeanTest.java:79]: These 4 lines are already in the other test, so removed them.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. I made a couple of minor changes (mentioned below) and a few cosmetic changes in the test before merging to trunk.

**Pull request closed without merging**